snooty/postprocess.py

```diff
@@ -123,6 +124,149 @@ def __call__(self, fileid_stack: FileIdStack, node: n.Node) -> None:
+                    target_identifier = node.children[0]
+                    if target_identifier.ids:
+                        return search_text in target_identifier.ids
+        return False
+
+    def bound_included_AST(
+        self,
+        nodes: MutableSequence[n.Node],
+        start_after_text: Optional[str],
+        end_before_text: Optional[str],
+    ) -> Tuple[MutableSequence[n.Node], bool, bool]:
+        """Given an AST in the form of nodes, return a subgraph of that AST by removing nodes 'outside' of
+            the bound formed by the nodes containing the start_after_text or end_before_text. In in-order traversal,
+            a node is considered 'outside' the subgraph if it precedes and is not any ancestor of the start-after node,
+            or if it succeeds and is not any ancestor of the end-before node."""
+
+        start_index, end_index = 0, len(nodes)
+        any_start, any_end = False, False
+
+        # For any given node: if the start_after node is within this node's subtree, do not include any
+        # preceding siblings of this node in the resulting AST; if the end_before node is within this
+        # node's subtree, then do not include any succeeding siblings of this node.
+        for i, node in enumerate(nodes):
+            has_start, has_end = False, False
+            # Determine if this node itself (not a child node) contains a bound
+            is_start = IncludeHandler.is_bound(node, start_after_text)
+            is_end = IncludeHandler.is_bound(node, end_before_text)
+            # Recursively search the child nodes for bounds
+            if isinstance(node, n.Parent):
+                children, has_start, has_end = self.bound_included_AST(
+                    node.children, start_after_text, end_before_text
+                )
+                node.children = children
+            if is_start or has_start:
+                any_start = True
+                start_index = i
+            if is_end or has_end:
+                any_end = True
+                end_index = i
+        if start_index > end_index:
+            raise Exception("start-after text should precede end-before text")
+        # Remove sibling nodes preceding and succeeding the nodes containing the bounds in their subtrees
+        return nodes[start_index : end_index + 1], any_start, any_end
+
+    def __call__(self, fileid_stack: FileIdStack, node: n.Node) -> None:
+        def get_include_argument(node: n.Directive) -> str:
+            """Get filename of include"""
+            argument_list = node.argument
+            assert len(argument_list) > 0
+            return argument_list[0].value
+
+        if not isinstance(node, n.Directive) or not node.name == "include":
+            return
+
+        argument = get_include_argument(node)
+        include_slug = clean_slug(argument)
+        include_fileid = self.slug_fileid_mapping.get(include_slug)
```

Comment:
Optional way to improve messy code that you didn't write: I think we can just use `FileId(include_slug)` rather than using this object as a lookup, as is done [here](https://github.com/mongodb/snooty-parser/blob/master/snooty/postprocess.py#L409).

But no need to block release if you don't want to do this!